src/services/AccommodationService.js

```diff
@@ -0,0 +1,86 @@
+import { likes, accommodations } from '../database/models';
+import CommonQueries from './CommonQueries';
+
+/**
+ * @export
+ * @class AccommodationService
+ */
+class AccommodationService {
+  /**
+ * user POST likes/dislikes accommodation facility
+ * @static
+ * @param {object} req pass request body
+ * @memberof AccommodationService
+ * @returns {object} either an error or data
+ */
+  static async addAccommodationLike(req) {
+    let setObject, displayMessage;
+    const { like } = req.query;
+    const { accommodationId } = req.params;
+    const { id } = req.user;
+
+    const accommodation = await CommonQueries.findOne(accommodations, { where: { id: accommodationId } });
+    const isLikedOrDisliked = await CommonQueries.findOne(likes, { where: { userId: id } });
+
+    if (isLikedOrDisliked) {
+      const { liked, disliked } = isLikedOrDisliked.dataValues;
+
+      if (like && liked && !disliked) {
+        setObject = { liked: false };
+        displayMessage = 'You unliked this accommodation!';
+      }
+      if (!like && !liked && disliked) {
+        setObject = { disliked: false };
+        displayMessage = 'You un-disliked this accommodation!';
+      }
+      if (!like && liked && !disliked) {
+        setObject = { disliked: true, liked: false };
+        displayMessage = 'You disliked this accommodation!';
+      }
+      if (!liked && !disliked) {
+        setObject = { liked: like, disliked: !like };
+        displayMessage = like ? 'You liked this accommodation!' : 'You disliked this accommodation!';
+      }
+
+      const updateLikeObject = [setObject, { where: { userId: id } }];
+
+      await CommonQueries.update(likes, updateLikeObject);
+    } else {
+      const addLikeObject = {
+        userId: id,
+        accommodationId,
+        liked: like,
+        disliked: !like
+      };
+      await CommonQueries.create(likes, addLikeObject);
+      displayMessage = like ? 'You liked this accommodation!' : 'You disliked this accommodation!';
+    }
+    return { displayMessage, accommodation };
+  }
+
+  /**
+   * user GET likes/dislikes accommodation facility
+   * @static
+   * @param {object} req pass request body
+   * @memberof AccommodationService
+   * @returns {object} either an error or data
+   */
+  static async getAccommodationLikes(req) {
+    const { accommodationId } = req.params;
+    const likeCounter = await CommonQueries.count(likes, {
+      where: {
+        accommodationId,
+        liked: true
+      },
+    });
+    const dislikeCounter = await CommonQueries.count(likes, {
+      where: {
+        accommodationId,
+        disliked: true
+      }
+    });
```

Comment:
@victorkarangwa4 What do you think if we save all common objects into one class so that whoever wants them, He/she can re-use them?

Reply:
Thank @NiyoEric for the feedback.
However, I've been trying to implement it, it's not working.
This is because a key can not be passed as a parameter in a function.
<img width="922" alt="Screen Shot 2019-12-06 at 08 15 46" src="https://user-images.githubusercontent.com/35301129/70300621-a1b0b300-1800-11ea-855c-1571dda5e8a2.png">